packages/ember-htmlbars/lib/keywords/closure-component.js

```diff
@@ -74,12 +74,14 @@ export function isComponentCell(component) {
 }
 
 function createNestedClosureComponentCell(componentCell, params, hash) {
+  let newHash = assign(Object.create(null), hash);
```

Comment:
I'll change it to `new EmptyObject()`. Thank you for the tip!

The thing is that processing positional params adds them to `hash` in place, therefore when component gets rerendered (or the component cell gets recomputed) the hash already has the value and the assertion fails. Other option would be to check that the param's name is not in the `hash` or, if it is, it is not exactly the same object.